### PR TITLE
Add documentation and refactored code

### PR DESCRIPTION
**Pull request by luii**
I've added some documentation to the code and tomorrow i add some stuff to the `README.md` so **dont** merge the PR yet

### CODE REVIEW

**Comment by piranna** [comment]
> I've added some documentation to the code and tomorrow i add some stuff to the README.md so dont merge the PR yet

ACK ;-) I was curious to know why did you forked the repos :-)

File: lib/getProcessStat.js [hunk 1]
```diff
@@ -26,9 +35,9 @@ function getProcessStat(pid, callback)
        'cguest_time', 'start_data', 'end_data', 'start_brk', 'arg_start',
        'arg_end', 'env_start', 'env_end', 'exit_code')
 
-   callback(null, result)
+   return callback(null, result)
```
Comment (piranna) [inline lib/getProcessStat.js:38]: Why the `return`? The function is not returning nothing at all, the fact of being used before is only just a convenience...

Comment (luii) [reply lib/getProcessStat.js:38]: The return Statement ends the execution in this code block so it wont execute anything after that, you're right that just a convenience but I think this should be done

**Comment by piranna** [comment]
I don't think so, return it's only used for early termination, the thing of returning the callback is just for doing it in one line.

**Comment by piranna** [comment]
Sorry for the delay. Due to the recent message, I've decided to merge your changes and take the opportunity to update the code :-)

**Pull request merged**